docs/src/2.5.AddingInteractivity.mdx

```diff
@@ -0,0 +1,136 @@
+import CodeSandboxEmbed from './jsx/utils/CodeSandboxEmbed.js';
+
+# Adding Interactivity
+
+In this chapter, you'll learn how to:
+
+- Set up the objects in the scene for mouse interaction
+- Interact with objects through event handlers
+- Retrieve object and event information from the event handlers
+
+## Adding and Removing Obstacles along the Duck's Path
+
+In previous chapters, you've learned how to interact with Worldview through mouse, keyboard, and programmatic controls, all of which affect the camera. Worldview and its [Command components](#/docs/api/command) also support other [mouse events](#/docs/api/mouse-events), such as `onClick`, `onDoubleClick`, `onMouseDown`, `onMouseMove`, and `onMouseUp`.
+
+In this step, let's make the spheres turn into large cubes when clicked. The duck will travel until it hits one of the cubes, and stop moving until you click again on the obstacle to remove it.
+
+First, let's add an `id` field to each sphere so that we can identify which one was clicked. The `id` value must be unique positive integers.
```

Comment:
```suggestion
First, let's add an `id` field to each sphere so that we can identify which one was clicked. The `id` value must be a unique positive integer.
```